src/test/scala/chiselTests/stage/WarningConfigurationSpec.scala

```diff
@@ -380,10 +380,10 @@ class WarningConfigurationSpec extends AnyFunSpec with Matchers with chiselTests
       e.getMessage should include("[W006] Cannot extract from Vec of size 0")
     }
 
-    it("should number AsTypeOfReadOnly as 8") {
-      val args = Array("--warn-conf", "id=8:e,any:s", "--throw-on-first-error")
+    it("should now error on AsTypeOfReadOnly") {
+      val args = Array("--throw-on-first-error")
       val e = the[Exception] thrownBy ChiselStage.emitCHIRRTL(new AsTypeOfReadOnly, args)
-      e.getMessage should include("[W008] Return values of asTypeOf will soon be read-only")
+      e.getMessage should include("Return values of asTypeOf are now read-only")
     }
```

Comment:
So this is the first time we'll be elevating a warning to an error. I haven't really worked through it all, we could add a corresponding `E008` for the old `W008`, but no other errors are numbered yet (and of course, unlike warnings, you will not be allowed to configure errors). Any thoughts?

Reply:
it seems like if we want to add error numbers we should do that more holistically? separately from this PR?